New packages: xplr-0.14.2

shubham-cpp opened the pull request:
<!-- Mark items with [x] where applicable -->

#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

<!--
If GitHub CI cannot be used to validate the build result (for example, if the
build is likely to take several hours), make sure to
[skip CI](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration).
When skipping CI, uncomment and fill out the following section.
Note: for builds that are likely to complete in less than 2 hours, it is not
acceptable to skip CI.
-->
#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (x86_64)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl

biopsin commented on srcpkgs/xplr/template:
```diff
@@ -0,0 +1,36 @@
+# distfiles="${homepage}/releases/download/v${version}/${pkgname}-linux.tar.gz"
+checksum=af7c7cb98a1e7b003c9ae1294fb7f7091ed48fe89ec01bfccac9aba154162c87
+
+do_fetch() {
+    mkdir ${wrksrc}
+    cd ${wrksrc}
+    curl -OLC - ${homepage}/releases/download/v${version}/${pkgname}-linux.tar.gz
+    curl -OLC - ${homepage}/raw/v${version}/LICENSE
+    curl -OLC - ${homepage}/raw/v${version}/README.md
+    curl -OLC - ${homepage}/raw/v${version}/src/init.lua
+    curl -OLC - ${homepage}/raw/v${version}/assets/desktop/${pkgname}.desktop
+}
```
Why curl and the "do_fetch" rutine and not let xbps-src just do its thing?

shubham-cpp replied:
The manual for xbps mentions that **distfiles** must end with archive format. Can we specify *.desktop,etc file in **distfiles** section

biopsin commented:
Also only one pkg per PR

Vaelatern commented:
This nowhere near meets quality requirements, not for commit names, not for package style, not for building, not for getting the distfiles, not for 1 package per PR (but acceptable in this case, please force push to this PR when you fix things, instead of losing history), nor for supported architectures.

shubham-cpp commented:
I didn't know there were so many requirements. I did look at [quality requirement](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality_requirements) section before hand

shubham-cpp commented:
Also I have closed the PR of xcolor since the author doesn't seem to create next release

Vaelatern commented:
In principle these packages can be merged.

However they can not be merged with these templates. Please fix the outstanding quality problems.

shubham-cpp commented:
> not for commit names, not for package style,

Didn't understand these

> not for building

By reading this [package-build-phases](https://github.com/void-linux/void-packages/blob/master/Manual.md#package-build-phases). I understand that I need to create `do_extract()` and `do_clean()`

> not for getting the distfiles, 

So I removed the `do_fetch()` and now it shows `ERROR: xplr-0.14.2_1: unknown distfile suffix for LICENSE.`
> not for 1 package per PR (but acceptable in this case, please force push to this PR when you fix things, instead of losing history), 

Fixed this by removing xcolor PR

> nor for supported architectures.

Can't verify this but will look to build on a 32-bit architecture using virtualbox

If there are resources I can look into to gain knowledge, please share. I'm a fan of void and would like to contribute as a gesture for appreciation of this distro

Vaelatern commented:
> > not for commit names, not for package style,
> 
> Didn't understand these

Commit names are documented at the bottom of the Manual.



> > not for building
> 
> By reading this [package-build-phases](https://github.com/void-linux/void-packages/blob/master/Manual.md#package-build-phases). I understand that I need to create `do_extract()` and `do_clean()`

No, you want to avoid creating any functions at all.


> > not for getting the distfiles,
> 
> So I removed the `do_fetch()` and now it shows `ERROR: xplr-0.14.2_1: unknown distfile suffix for LICENSE.`

Better than having the `do_fetch`. This error can be fixed.

> > not for 1 package per PR (but acceptable in this case, please force push to this PR when you fix things, instead of losing history),
> 
> Fixed this by removing xcolor PR

Acceptable, thank you for doing this and staying within this PR.

> > nor for supported architectures.
> 
> Can't verify this but will look to build on a 32-bit architecture using virtualbox

`xbps-src` will cross build if you tell it to.

> If there are resources I can look into to gain knowledge, please share. I'm a fan of void and would like to contribute as a gesture for appreciation of this distro

The Manual is a great source. Between the Manual and README, you should have all the reference material necessary. You can also look at other rust packages to see how it's done there.

shubham-cpp commented:
1. Removed do_fetch
2. Added tags variable
3. Updated checksums

Also used `xbps-src pkg -a i686 xplr` and it built successfully but it failed for `-a armv7l` and `-a aarch64`. 
Just gave me error that it couldn't built for  armv7l/aarch64 nothing more than that.

The only issue now is regarding 
> commit names

So should I create a new PR(as mentioned in [continuous-integration](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration)) or will this be accepted?

Vaelatern commented:
Please do not create a new PR.

Remove `archs` line and try cross building again.

shubham-cpp commented:
Ok noted 📝

shubham-cpp commented:
So I ran `xbps-src pkg -f -a <arch_name> -K xplr` on 
- x86_64* 
- i686*
-  aarch64* 
- armv[67]l* 
- armv[67]hf*

The command did gave me following errors errors
- ERROR: [reposync] failed to fetch file `https://alpha.de.repo.voidlinux.org/current/musl/ppc64-musl-repodata':
- ERROR: base-files-0.142_11: [unpack] failed to extract file `./bin': Directory not empty

For following architectures  
- armv5te
- mips-musl
- ppc64
- ppcle

Vaelatern commented:
Do you still have your own `do_extract`?

Stop that.

biopsin commented on srcpkgs/xplr/template:
```diff
@@ -0,0 +1,43 @@
+license="MIT"
+homepage="https://github.com/sayanarijit/xplr"
+distfiles="${homepage}/releases/download/v${version}/${pkgname}-linux.tar.gz
+	${homepage}/raw/v${version}/LICENSE
+	${homepage}/raw/v${version}/README.md
+	${homepage}/raw/v${version}/src/init.lua
+	${homepage}/raw/v${version}/assets/desktop/${pkgname}.desktop"
```
Drop.
These files are all present in the source pkg and therefore also correct checksum to only one ..

shubham-cpp replied:
The [xplr](https://github.com/sayanarijit/xplr/releases/download/v0.14.2/xplr-linux.tar.gz) release includes only binary and noting else. I have used [PKGBUILD](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=xplr-bin) as reference

Vaelatern replied:
and no source code? You can't get the source code somehow?

Vaelatern replied:
PKGBUILDs, apparently, are not a good reference for Void packages. We prefer building from source. Please correct this defect in these templates.

shubham-cpp replied:
Bare with my ignorance, but the whole point of using void is to have binary packages and compile less as compared to arch, etc

ericonr replied:
Binary packages are still built from some source. We prefer to be the ones to build from source, not just repackage files provided by upstream

shubham-cpp replied:
Then I don't think we need this template. Because we could just install xplr using cargo.
